packages/admin/admin/src/router/PromptHandler.tsx

```diff
@@ -115,18 +112,13 @@ export const RouterPromptHandler: React.FunctionComponent<Props> = ({ children,
     }: {
         id: string;
         message: (location: History.Location, action: History.Action) => string | boolean;
-        saveAction: SaveAction;
+        saveAction?: SaveAction;
         path: string;
         subRoutePath?: string;
     }) => {
-        registeredMessages.current[id] = { message, path, subRoutePath };
-        if (saveAction) {
-            saveActions.current[id] = saveAction;
-        }
+        registeredMessages.current[id] = { message, path, subRoutePath, saveAction };
         // If saveAction is passed it has to be passed for all registered components
-        const countSaveActions = Object.keys(saveActions.current).length;
-        const countRegisteredMessages = Object.keys(registeredMessages.current).length;
-        if (countSaveActions > 0 && countSaveActions !== countRegisteredMessages) {
+        if (saveAction && Object.values(registeredMessages.current).some((registeredMessage) => !registeredMessage.saveAction)) {
```

Comment:
I don't really understand this `if`. Will there ever be multiple save actions? Is this legacy code?

Reply:
There can be multiple `<Prompt`s on one page and `saveAction` is optional. When no Prompt gives a saveAction the save button is now shown and everything is fine. When all Promp have a saveAction, the save button is shown any everything will be saved - also fine.

But when one Prompt is dirty and has no saveAction and another Prompt is also dirty but has a saveAction we also show the save button and can't save the first Prompt.

(this logic was like that before, I did only refactor it)

Reply:
Thanks for the explanation!